angular/src/directives/navigation/ion-router-outlet.ts

```diff
@@ -240,8 +238,33 @@ export class IonRouterOutlet implements OnDestroy, OnInit {
       const activatedRouteProxy = this.createActivatedRouteProxy(component$, activatedRoute);
 
       const injector = new OutletInjector(activatedRouteProxy, childContexts, this.location.injector);
-      cmpRef = this.activated = this.location.createComponent(factory, this.location.length, injector);
 
+      if (resolverOrInjector && isComponentFactoryResolver(resolverOrInjector)) {
+        // Backwards compatibility for Angular 13 and lower
+        const factory = resolverOrInjector.resolveComponentFactory(component);
+        cmpRef = this.activated = this.location.createComponent(factory, this.location.length, injector);
+      } else {
+        /**
+         * Angular 14 and higher.
+         *
+         * When we drop < Angular 14, we can replace the following code with:
```

Comment:
Could we add a tech debt ticket here too? (Could be the same ticket as the other one)

Also, if/when we don't support Angular 13, do we still need the backwards compatibility code above?